### PR TITLE
Use full DocWriteResponse json for Put/Update/Delete responses

### PR DESCRIPTION
**Pull request by dbwiddis**
### Description

Updates the SDKClient to use the full IndexResponse, UpdateResponse, and DeleteResponse objects.

Parallel of similar work on GetResponse in #2587 
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/src/test/java/org/opensearch/ml/model/MLModelGroupManagerTests.java [hunk 1]
```diff
@@ -198,7 +196,7 @@ public void test_ModelGroupNameNotUnique() throws IOException, InterruptedExcept
         CountDownLatch latch = new CountDownLatch(1);
         LatchedActionListener<String> latchedActionListener = new LatchedActionListener<>(actionListener, latch);
         mlModelGroupManager.createModelGroup(mlRegisterModelGroupInput, latchedActionListener);
-        latch.await();
+        latch.await(500, TimeUnit.MILLISECONDS);
```
Comment (dhrubo-os) [inline plugin/src/test/java/org/opensearch/ml/model/MLModelGroupManagerTests.java:199]: Why do we need 500 now?

Comment (dhrubo-os) [reply plugin/src/test/java/org/opensearch/ml/model/MLModelGroupManagerTests.java:199]: What happens after that? Will there by any chance for these tests to be flaky if timeout happens?

Comment (dbwiddis) [reply plugin/src/test/java/org/opensearch/ml/model/MLModelGroupManagerTests.java:199]: The purpose for the latch is a race condition as the async stuff processes in multiple threads.  

The previous tests returned immediately in the same thread with a mock `.when()` response.  Now there is an async execution layer in between the call and the mock so there are two thread involved.

It should complete in sub-millisecond time in most cases, no more than a few milliseconds in nearly all cases.

Leaving everything as `latch.await()` with no timeout would frequently stall tests and have them not complete until the test suite itself killed the threads on timeout -- this worked on CI but was frustrating debugging local test failures when a test just kept running and running.

TLDR: we'll only hit the timeout if there's a bug in the code, and it will help get a stack trace faster to make debugging easier and save developers like me a ton of debugging time.  Worst case scenario if CI is running on a fully loaded machine and multithreaded tests take more than half a second to run, they'll fail flakily, but never pass when they should have failed.

**Review by dhrubo-os — APPROVED** [review]
